parity/cli/mod.rs

```diff
@@ -1816,8 +1815,8 @@ mod tests {
 			arg_jsonrpc_cors: "null".into(),
 			arg_jsonrpc_apis: "web3,eth,net,parity,traces,rpc,secretstore".into(),
 			arg_jsonrpc_hosts: "none".into(),
-			arg_jsonrpc_server_threads: None,
-			arg_jsonrpc_threads: 4,
+			arg_jsonrpc_server_threads: Some(4),
+			arg_jsonrpc_threads: 0, // DEPRECATED, does nothing
```

Comment:
Shall we print a warning that this setting is going to be removed in further versions?

Reply:
There's [this](https://github.com/paritytech/parity-ethereum/pull/11267/files#diff-54d6bbd5addb7a0d39c647e4e33e6990R509-R511) but maybe we can do more to make it stand out? 
@joshua-mir @Tbaut do you guys remember how we dealt with deprecation in the past?

Reply:
Well, that's only in docs. What I mean is to print a warning if someone is using that parameter (either CLI or config) - now we just ignore it, but if we decide to remove it the app will not run for them due to this extra CLI flag or config entry. I believe we had cases in the past where this grace period is introduced, before the flag is removed completely.